test/test_ufbrw.F90

```diff
@@ -0,0 +1,180 @@
+      data brr(11) /"300.00 ******** ******** ******** ********"/
+      data brr(12) /""/
+      data brr(13) /"POB<1000 UOB<1000"/
+      data brr(14) /"985.20 ******** ********    -2.80    -7.70"/
+      data brr(15) /"998.00 ******** ********     4.60     2.20"/
+      data brr(16) /"850.00 ******** ********     2.00    -1.70"/
+      data brr(17) /"700.00 ******** ********    12.10    -4.40"/
+      data brr(18) /""/
+      data brr(19) /"POB<800  TOB<1000"/
+      data brr(20) /"700.00  2131.00     9.20 ******** ********"/
+      data brr(21) /""/
+      data brr(22) /"POB<851  POB>699"/
+      data brr(23) /"850.00 ******** ********     2.00    -1.70"/
+      data brr(24) /"700.00 ******** ********    12.10    -4.40"/
+      data brr(25) /"850.00  2161.00    14.80 ******** ********"/
+      data brr(26) /"700.00  2131.00     9.20 ******** ********"/
+      data brr(27) /""/
+      data brr(28) /"POB=850"/
+      data brr(29) /"850.00 ******** ********     2.00    -1.70"/
+      data brr(30) /"850.00  2161.00    14.80 ******** ********"/
+      data brr(31) /""/
+      data brr(32) /""/
+      data brr(33) /"read/write from unit 20"/
+      data brr(34) /"985.20 ******** ********    -2.80    -7.70"/
+      data brr(35) /"985.20 12968.00    31.30 ******** ********"/
+      data brr(36) /"998.00 ******** ********     4.60     2.20"/
+      data brr(37) /"850.00 ******** ********     2.00    -1.70"/
+      data brr(38) /"700.00 ******** ********    12.10    -4.40"/
+      data brr(39) /"998.00  8112.00    22.30 ******** ********"/
+      data brr(40) /"925.00  6312.00    14.10 ******** ********"/
+      data brr(41) /"850.00  2161.00    14.80 ******** ********"/
+      data brr(42) /"700.00  2131.00     9.20 ******** ********"/
+      data brr(43) /"300.00 ******** ******** ******** ********"/
+      data brr(44) /""/
+      data brr(45) /"read/write from unit 50"/
+      data brr(46) /"985.20 ******** ********    -2.80    -7.70"/
+      data brr(47) /"985.20 12968.00    31.30 ******** ********"/
+      data brr(48) /"998.00 ******** ********     4.60     2.20"/
+      data brr(49) /"850.00 ******** ********     2.00    -1.70"/
+      data brr(50) /"700.00 ******** ********    12.10    -4.40"/
+      data brr(51) /"998.00  8112.00    22.30 ******** ********"/
+      data brr(52) /"925.00  6312.00    14.10 ******** ********"/
+      data brr(53) /"850.00  2161.00    14.80 ******** ********"/
+      data brr(54) /"700.00  2131.00     9.20 ******** ********"/
+      data brr(55) /"300.00 ******** ******** ******** ********"/
+
+      print *, 'Testing misc subroutines.'
+
+#ifdef KIND_8
+      call setim8b(.true.)
+#endif
+
+! set the name of the bufr testfile to process
+
+      file="testfiles/data/prepbufr2.ref"
+
+! open the print file to record ufbrep test results
+
+      open(55,file='ufbrw_prnt_out')
+
+! test various reading options applying user specified filtering, testing conwin
+
+      do i=1,5
```

Comment:
Does your editor not do autoindenting? I would expect indenting inside do loops...

Reply:
I prefer straight edges but did the indents anyway.